giraffe/src/utils/formatters.test.ts

```diff
@@ -104,6 +104,26 @@ describe('timeFormatter', () => {
     )
   })
 
+  test('24 hour format and UTC format should not show "24" for midnight', () => {
+    const utcFormatterWithFormat = timeFormatter({
+      timeZone: 'UTC',
+      format: 'YYYY-MM-DD HH:mm:ss ZZ',
+    })
+    const nonUTCFormatterWithFormat = timeFormatter({
+      timeZone: 'Europe/Berlin',
+      format: 'YYYY-MM-DD HH:mm:ss ZZ',
+      hour12: false,
+    })
+
+    let midnight = new Date('2019-01-01T00:00Z')
+    expect(utcFormatterWithFormat(midnight)).toEqual('2019-01-01 00:00:00 UTC')
+
+    midnight = new Date('2019-01-01T23:00Z')
```

Comment:
can we add another test case for midnight being `24`?

can we also add a test case for normal boring cases, for before noon and after noon? e.g. for when the time is like 9 am and when the time is 3 pm?

Reply:
1. That is very insightful suggestion. I didn't even realize `Date` can accept "24" in the hours. It can. And it is exactly the kind of test case that addresses this issue. "24" is technically allowed, even though it looks "bad" or "embarrassing" in a time series database.

2. Sure, will do.